prac_08/convert_miles_km.py

```diff
@@ -1,29 +1,41 @@
 from kivy.app import App
 from kivy.lang import Builder
 from kivy.core.window import Window
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.properties import StringProperty
 
 __author__ = 'Tahlia Clennar'
 
+CONVERSION = 1.61
+
 
 class UnitConversionApp(App):
+    """Convert miles to kilometers in GUI"""
+    kilometers = StringProperty()
 
     def build(self):
         """ build the Kivy app from the kv file """
-        Window.size = (200, 100)
         self.title = "Convert Miles to Kilometers"
         self.root = Builder.load_file('convert_miles_km.kv')
         return self.root
 
     def handle_calculate(self, value):
         """ handle calculation (could be button press or other call), output result to label widget """
+        miles = self.convert_to_number(value)
+        self.kilometers = str(miles * CONVERSION)
```

Comment:
Rather than creating another function to convert from string to float/int, you could put the same try block in this function and update the result as a string to your dynamic `StringProperty()`.  Something like:
```    
def handle_calculate(self, value): 
       """Converts Miles to Kilometers"""
        try:
            result = float(value) * CONVERSION
            self.kilometers = str(f"{result:.3f}")
        except ValueError:
            self.kilometers = str("0.0") 
```